onnxruntime/core/graph/featurizers_ops/featurizers_defs.cc

```diff
@@ -1656,6 +1687,104 @@ void RegisterRollingWindowFeaturizerVer1() {
+      //     +===========+=======+===================+
+      //     | A         | 10    | [[NAN]]           |
+      //     +-----------+-------+-------------------+
+      //     | A         | 4     | [[10]]            |
+      //     +-----------+-------+-------------------+
+      //     | A         | 6     | [[4]]             |
+      //     +-----------+-------+-------------------+
+      //     | A         | 11    | [[4]]             |
+      //     +-----------+-------+-------------------+
+      //     A more complex example would be, assuming we have horizon = 2, maxWindowSize = 2, minWindowSize = 2, and we want the maximum value
+      //     +-----------+-------+-------------------+
+      //     | grain     | target| target_max        |
+      //     +===========+=======+===================+
+      //     | A         | 10    | [[NAN, NAN]]      |
+      //     +-----------+-------+-------------------+
+      //     | A         | 4     | [[NAN, NAN]]      |
+      //     +-----------+-------+-------------------+
+      //     | A         | 6     | [[NAN, 10]]       |
+      //     +-----------+-------+-------------------+
+      //     | A         | 11    | [[10, 6]]         |
+      //     +-----------+-------+-------------------+
+  //)DOC";
+
+  MS_FEATURIZERS_OPERATOR_SCHEMA(SimpleRollingWindowTransformer)
+      .SinceVersion(1)
+      .SetDomain(kMSFeaturizersDomain)
+      .Input(
+          0,
+          "State",
+          "State generated during training that is used for prediction",
+          "T0")
+      .Input(
+          1,
+          "Grains",
+          "Grains tensor of shape [R][K].",
+          "GrainT")
+      .Input(
+          2,
+          "Target",
+          "Target tensor of shape [R]",
+          "T")
+      .Output(
+          0,
+          "Output",
+          "Output tensor of shape [R][M]",
+          "OutputT")
+      .TypeConstraint(
+          "T0",
+          {"tensor(uint8)"},
+          "No information is available")
+      .TypeConstraint(
+          "GrainT",
+          {"tensor(string)"},
+          "No information is available")
+      .TypeConstraint(
+          "T",
+          {"tensor(float)", "tensor(double)"},
+          "No information is available")
+      .TypeConstraint(
+          "OutputT",
+          {"tensor(double)"},
+          "No information is available")
+      .TypeAndShapeInferenceFunction(
+          [](ONNX_NAMESPACE::InferenceContext& ctx) {
+            propagateElemTypeFromDtypeToOutput(ctx, ONNX_NAMESPACE::TensorProto_DataType_DOUBLE, 0);
+            if (hasInputShape(ctx, 1)) {
+              const auto& grains_shape = getInputShape(ctx, 1);
+              if (grains_shape.dim_size() != 2) {
+                fail_shape_inference("Expecting Grains to have 2 dimensions");
+              }
+
+              ONNX_NAMESPACE::TensorShapeProto shape;
+              *shape.add_dim() = grains_shape.dim(0);
+              shape.add_dim();
+              shape.add_dim();
```

Comment:
Should these 2 lines be duplicated?

Reply:
Since the output is 3 dimensional, the two same lines are adding the 2nd and 3rd dimension